legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-bigquery/src/main/java/org/finos/legend/engine/persistence/components/relational/bigquery/sqldom/schemaops/statements/CopyStatement.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.persistence.components.relational.bigquery.sqldom.schemaops.statements;
+
+import org.finos.legend.engine.persistence.components.relational.sqldom.SqlDomException;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schemaops.expresssions.table.Table;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schemaops.statements.DMLStatement;
+
+import static org.finos.legend.engine.persistence.components.relational.sqldom.common.Clause.LOAD_DATA;
+import static org.finos.legend.engine.persistence.components.relational.sqldom.common.Clause.OVERWRITE;
+import static org.finos.legend.engine.persistence.components.relational.sqldom.utils.SqlGenUtils.WHITE_SPACE;
+
+public class CopyStatement implements DMLStatement
+{
+    private Table table;
+    private SelectFromFileStatement selectFromFileStatement;
+
+    public CopyStatement()
+    {
+    }
+
+    /*
+     Copy GENERIC PLAN for Big Query:
+        LOAD DATA OVERWRITE table_name
+        (COLUMN_LIST)
+        FROM FILES (LOAD_OPTIONS)
```

Comment:
We can handle COLUMN_LIST under this class
FILES (LOAD_OPTIONS) can be handled as a TableType

Reply:
Refactored